include/oneapi/tbb/blocked_rangeNd.h

```diff
@@ -100,17 +110,23 @@ class blocked_rangeNd_impl<Value, N, detail::index_sequence<Is...>> {
 
     blocked_rangeNd_impl(blocked_rangeNd_impl& r, proportional_split proportion) : my_dims(r.my_dims) {
         do_split(r, proportion);
+        __TBB_ASSERT(my_dims.size()==N, "range is not properly initialised" );
     }
 
     blocked_rangeNd_impl(blocked_rangeNd_impl& r, split proportion) : my_dims(r.my_dims) {
         do_split(r, proportion);
+        __TBB_ASSERT(my_dims.size()==N, "range is not properly initialised" );
     }
 
 private:
     static_assert(N != 0, "zero dimensional blocked_rangeNd can't be constructed");
 
-    //! Ranges in each dimension.
-    std::array<tbb::blocked_range<value_type>, N> my_dims;
+    /*! Ranges in each dimension.
+     *
+     * This has to be a vector, as there are situations where we can't
+     * initialise the entries all in one go.
+     */
+    std::vector<tbb::blocked_range<value_type>> my_dims;
```

Comment:
The change to vector might not be necessary, as I think the constructor can be written so that the array is properly initialized - see another comment.